registration/tests/test_views.py

```diff
@@ -589,7 +593,7 @@ def test_modify_user(self):
         application = BetaTestApplication.objects.get()
         self._assert_application_matches_form_data(application)
 
-    @override_settings(VARIABLES_NOTIFICATION_EMAIL='notifications@opencraft.com')
+    @override_settings(VARIABLES_NOTIFICATION_EMAIL='notifications@opencraft.com', INSTANCE_STORAGE_TYPE='s3')
```

Comment:
@pkulkark, though the unit tests on this PR and the merge commit passed, this change is now causing this unit test to fail with the below error.

```
======================================================================
ERROR: test_modifying_design_fields_sends_email (registration.tests.test_views.BetaTestApplicationViewTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/vagrant/.pyenv/versions/opencraft/lib/python3.6/site-packages/django/test/utils.py", line 373, in inner
    return func(*args, **kwargs)
  File "/home/vagrant/opencraft/registration/tests/test_views.py", line 632, in test_modifying_design_fields_sends_email
    EmailAddress.objects.confirm(email.key)
  File "/home/vagrant/.pyenv/versions/opencraft/lib/python3.6/site-packages/simple_email_confirmation/models.py", line 203, in confirm
    email_confirmed.send(sender=address.user, email=address.email)
  File "/home/vagrant/.pyenv/versions/opencraft/lib/python3.6/site-packages/django/dispatch/dispatcher.py", line 175, in send
    for receiver in self._live_receivers(sender)
  File "/home/vagrant/.pyenv/versions/opencraft/lib/python3.6/site-packages/django/dispatch/dispatcher.py", line 175, in <listcomp>
    for receiver in self._live_receivers(sender)
  File "/home/vagrant/opencraft/registration/provision.py", line 51, in provision_instance
    _provision_instance(sender, **kwargs)
  File "/home/vagrant/opencraft/registration/provision.py", line 79, in _provision_instance
    application.instance.lms_users.add(user)
AttributeError: 'NoneType' object has no attribute 'lms_users'

----------------------------------------------------------------------
```
Removing the `INSTANCE_STORAGE_TYPE='s3' setting fixes it. Can you check?

Reply:
The issue is happening because of the check [here](https://github.com/open-craft/opencraft/blob/master/instance/factories.py#L47) causing `None` to be returned by the `production_instance_factory` function instead of an instance. So we should remove this.

Reply:
@lgp171188 I see. But reverting this would result in the CI failure again right? I can see this test case fails in the [example build](https://app.circleci.com/jobs/github/open-craft/opencraft/11978) that was failing before this change. Is this error seen in the devstack? I don't see it failing on a [recent unit test run](https://circleci.com/gh/open-craft/opencraft/12786?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)

Reply:
@pkulkark, even I am not sure why and how the unit tests for this PR passed. This unit test is failing in the devstack and also on CircleCI (see https://github.com/open-craft/opencraft/pull/530).

Reply:
@lgp171188 Interesting. I'll raise another PR which reverts the override for this test case and see if the CI still passes.